external-crates/move/crates/move-model-2/src/summary.rs

```diff
@@ -688,8 +688,15 @@ impl Fields {
             N::StructFields::Native(_) => return,
         };
         self.positional_fields = is_positional;
+        let pos_name_of = |sym| {
```

Comment:
This fixes a bug when annotating positional fields where the field name is `pos0` but the name in the source annotations is `0`.

Reply:
I wonder if we should change this in the model too? or just keep it as is in the programinfo?

Reply:
I was on the fence on this, however I think it makes sense to do this here as opposed to pushing it into the model. My reasoning being that in a bytecode-model we don't _know_ that a field is positional or not (or if someone just named their field pos0...), so introducing this distinction would just introduce some drift between binary and source models that don't really need to be there.